src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -118,6 +118,8 @@ class IAggregateFunction : public std::enable_shared_from_this<IAggregateFunctio
     /// window function.
     virtual void insertResultInto(AggregateDataPtr __restrict place, IColumn & to, Arena * arena) const = 0;
 
+    virtual void insertResultIntoAndDestroyStatesBatch(size_t batch_size, AggregateDataPtr * places, size_t offset, IColumn & to, Arena * arena) const = 0;
```

Comment:
The whole idea will not work, because we have to destroy the states of several aggregate functions and zero-out the pointer in hash table in exceptional-safe way.

Reply:
If we will find a workaround it will be great!